### ISSUE TITLE
Using InitSlicedFindPath with FindPathAnyAngle results in a crash in UpdateSlicedFindPath

### ISSUE THREAD

**Issue by CodeSpartan** [arl/go-detour#24]
Looking at the callstack, the problem stems from here: 
_go-detour/detour/query.go:1634_ - **lastPos** is nil
It's declared, but never allocated. So _copy(lastPos, curPos[:3])_ (line 1620) doesn't know where to copy to, since lastPos is still nil. Adding _lastPos = d3.NewVec3()_ before the copy solves the issue.

**Comment by arl** [comment]
Hi. Thank you for that.
I guess we're missing a test somewhere.
Since you solved the issue, would you care to open a PR?

**Comment by CodeSpartan** [comment]
Hi! Yes, I'll try to do that soon. 

Am I correct in understanding that you only ported _SlicedFindPath_ and its related methods (_update_, _finalize_, etc), but haven't tested them? I'm guessing that you only used regular _findpath_ in your game project? I've found multiple crashes and errors when using sliced paths. I fixed what problems I stumbled on, but I'm still a bit apprehensive of using this library for fear of catching tough to fix errors down the line. A warning in the readme about Sliced paths not having been tested would've helped.

Other than that, great project. Without it, Go would have no pathfinding solutions whatsoever.

I'm also going to try to port the 64 bit version from the original project. They do it using a **#define DT_POLYREF64 1** macro, but go doesn't have macros, so I'll push it into a separate branch. Useful for large worlds, apparently, though how large it's not specified, which is a shame.
